cmd/copy.go

```diff
@@ -948,6 +1025,136 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	return nil
 }
 
+// TODO: include this within initResourceTraverser?
+func initPipeline(ctx context.Context, location common.Location, credential common.CredentialInfo) (p pipeline.Pipeline, err error) {
+	switch location {
+	case common.ELocation.Local():
+		return nil, nil
+	case common.ELocation.Blob():
+		p, err = createBlobPipeline(ctx, credential)
+	case common.ELocation.File():
+		p, err = createFilePipeline(ctx, credential)
+	case common.ELocation.BlobFS():
+		p, err = createBlobFSPipeline(ctx, credential)
+	default:
+		err = fmt.Errorf("can't produce new pipeline for location %s", location)
+	}
+
+	return
+}
+
+func trimWildcards(path string) string {
+	if strings.Index(path, "*") == -1 {
+		return path
+	}
+
+	firstWCIndex := strings.Index(path, "*")
+	result := replacePathSeparators(path[:firstWCIndex])
+	lastSepIndex := strings.LastIndex(result, "/")
+	result = result[:lastSepIndex+1]
+
+	return result
+}
+
+func (cca *cookedCopyCmdArgs) findObject(source bool, dstIsDir bool, object storedObject) (relativePath string) {
+	var pathEncodeRules = func(path string) string {
+		loc := common.ELocation.Unknown()
+
+		if source {
+			loc = cca.fromTo.From()
+		} else {
+			loc = cca.fromTo.To()
+		}
+		pathParts := strings.Split(path, "/")
```

Comment:
Right, pardon me.

Reply:
I don't know for sure, but I believe that was the intention. In general it's good practice to avoid string literals which are annoying to find/replace in case of changes. 😄